### PR TITLE
Kria kv260

### PR DESCRIPTION
**Pull request by igoropaniuk**

### CODE REVIEW

**Review by vanmaegima — APPROVED** [review]
LGTM

**Review by MrCry0 — APPROVED** [review]
LGTM

File: meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common.cmd.in [hunk 1]
```diff
@@ -152,19 +152,25 @@ if test "${fiovb.is_secondary_boot}" = "0"; then
...
+			echo "${fio_msg} update primary boot path with validated images ..."
 
-		setenv image_path "${ostree_root}/usr/lib/firmware/${bootloader_image}"
-		setenv start_blk "${bootloader}"
-		run load_image
-		run update_primary_image
+			setenv image_path "${ostree_root}/usr/lib/firmware/${bootloader_image}"
+			setenv start_blk "${bootloader}"
+			run load_image
+			run update_primary_image
```
Comment (angolini) [inline meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common.cmd.in:161]: Could you, please, take a look why github highlight lines 158-161, but at the same time does not highlight 163-166?

**Review by angolini — APPROVED** [review]
only one minor comment

File: meta-lmp-bsp/dynamic-layers/xilinx/recipes-bsp/u-boot/u-boot-xlnx_2022.01.bb [hunk 2]
```diff
@@ -3,7 +3,7 @@ UBOOT_VERSION = "v2022.01"
 UBRANCH = "xilinx-v2022.01-rebase"
 UBOOTURI = "git://github.com/foundriesio/u-boot.git;protocol=https"
 
-SRCREV = "a056b687f497022ad2aeaf40ef44d550448c31ff"
+SRCREV = "49aea78e3033e9d78108f7d12cc98ba701632f4a"
```
Comment (ricardosalveti) [inline meta-lmp-bsp/dynamic-layers/xilinx/recipes-bsp/u-boot/u-boot-xlnx_2022.01.bb:6]: We should probably bump to the latest now, including your latest pr that is now approved.

File: meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common.cmd.in [hunk 3]
```diff
@@ -175,7 +175,7 @@ fi
 
 setenv bootcmd_load_f 'ext4load ${devtype} ${devnum}:${rootpart} ${fit_addr} "/boot"${kernel_image}'
 setenv bootcmd_tee_ovy 'if test -n ${optee_ovl_addr}; then imxtract ${fit_addr}#conf@@FIT_NODE_SEPARATOR@@${fdt_file_final} fdt@@FIT_NODE_SEPARATOR@@${fdt_file_final} ${fdt_addr}; fdt addr ${fdt_addr}; fdt resize 0x1000; fdt apply ${optee_ovl_addr}; fi'
-setenv bootcmd_run 'if test -n ${optee_ovl_addr}; then bootm ${fit_addr}#conf@@FIT_NODE_SEPARATOR@@${fdt_file_final} ${fit_addr}#conf@@FIT_NODE_SEPARATOR@@${fdt_file_final} ${fdt_addr}; else echo "${fio_msg} loading ${fdt_file_final}"; bootm ${fit_addr}#conf@@FIT_NODE_SEPARATOR@@${fdt_file_final}; fi'
+if test -z "${bootcmd_run}"; then setenv bootcmd_run 'if test -n ${optee_ovl_addr}; then bootm ${fit_addr}#conf@@FIT_NODE_SEPARATOR@@${fdt_file_final} ${fit_addr}#conf@@FIT_NODE_SEPARATOR@@${fdt_file_final} ${fdt_addr}; else echo "${fio_msg} loading ${fdt_file_final}"; bootm ${fit_addr}#conf@@FIT_NODE_SEPARATOR@@${fdt_file_final}; fi'; fi
```
Comment (ricardosalveti) [inline meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common.cmd.in:191]: Won't this cause a problem when the env gets saved and reused on a following boot? If we end up reusing the environment that was previously saved we might not necessarily be able to update the bootcmd run content via OTAs.

**Comment by ricardosalveti** [comment]
"bsp: u-boot-xlnx: k62: add qspi configuration" -> it should be k26 instead.

File: meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common.cmd.in (hunk 1 above)
Comment (ricardosalveti) [reply meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common.cmd.in:161]: Funny as it all seems to be just tabs.

File: meta-lmp-bsp/dynamic-layers/xilinx/recipes-bsp/arm-trusted-firmware/arm-trusted-firmware_2021.2.bbappend [hunk 4]
```diff
@@ -2,5 +2,5 @@
 PROVIDES += "virtual/trusted-firmware-a"
 
 # Enable opteed as the main SPD provider (required for optee)
-EXTRA_OEMAKE:append:zynqmp = " SPD=opteed"
+EXTRA_OEMAKE:append:uz3eg-iocc = " SPD=opteed"
```
Comment (ricardosalveti) [inline meta-lmp-bsp/dynamic-layers/xilinx/recipes-bsp/arm-trusted-firmware/arm-trusted-firmware_2021.2.bbappend:5]: Please change to 'uz' as we have other customer specific machines that are uz-based that would also be affected by this change.

File: meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common.cmd.in (hunk 3 above)
Comment (MrCry0) [reply meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common.cmd.in:191]: +1
It's better to check a different env variable and drop it before saving env. Like
```
if test -z "${bootcmd_custom_run}"'; then ...; else setenv bootcmd_run ${bootcmd_custom_run}; setenv bootcmd_custom_run; fi
```

File: meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common.cmd.in (hunk 1 above)
Comment (igoropaniuk) [reply meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common.cmd.in:161]: that's just tabs

**Comment by igoropaniuk** [comment]
all comments addressed

**Review by quaresmajose — COMMENTED** [review]
lgtm but I don't have too much context on the uboot scripts

**Review by ricardosalveti — APPROVED** [review]
LGTM

**Pull request merged**